src/value/use-on-change.ts

```diff
@@ -6,15 +6,14 @@ export function useOnChange<T>(
     value: MotionValue<T> | number | string,
     callback: Subscriber<T>
 ) {
-    useEffect(
-        () => (isMotionValue(value) ? value.onChange(callback) : undefined),
-        [value]
+    useEffect(() =>
+        isMotionValue(value) ? value.onChange(callback) : undefined
```

Comment:
Is this on change returning the unsubscribe handler?

Reply:
Yeah